docs/docs/examples/workflow/multi_step_query_engine.ipynb

```diff
@@ -0,0 +1,493 @@
+{
```

Comment:
<pre class="ql-syntax" spellcheck="false">Line #75.                    prev_reasoning,
</pre><p>The first time this loop runs, <code>prev_reasoning</code> will be empty. Should we just skip calling this function in that case and use the original query bundle? </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/run-llama/llama_index/pull/15438/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='9'/>

Reply:
No, we cannot skip it because for the first time as well it will create a sub query. Here is an example for the same:

```
original_query = "In which city did the author found his first company, Viaweb?"

**Loop-1:**
previous reasoning: "" 
updated_query_bundle: Who is the author who founded Viaweb?

**Loop-2:**

previous reasoning: - Who is the author who founded Viaweb?
- The author who founded Viaweb is Paul Graham.

updated_query_bundle: In which city did Paul Graham found his first company, Viaweb?

**Loop-3:**

previous reasoning: - Who is the author who founded Viaweb?
- The author who founded Viaweb is Paul Graham.
- In which city did Paul Graham found his first company, Viaweb?
- Paul Graham founded his first company, Viaweb, in Cambridge.

updated_query_bundle: None

-----STOPS as query_bundle is None-----
```